plugin/plugin.go

```diff
@@ -365,3 +365,74 @@ func (p *Plugin) getMessage(t string) *descriptor.DescriptorProto {
 	return file.GetMessage(strings.TrimPrefix(t, "."+file.GetPackage()+"."))
 }
 
+func (p *Plugin) GetDeclinedMethods(options []av_opts.AtlasValidateFieldOption_Operation) []string {
+	httpMethods := make(map[string]struct{}, 0)
+	for _, op := range options {
+		switch op {
+		case av_opts.AtlasValidateFieldOption_create:
+			httpMethods["POST"] = struct{}{}
+		case av_opts.AtlasValidateFieldOption_update:
+			httpMethods["PUT"] = struct{}{}
+			httpMethods["PATCH"] = struct{}{}
+		}
+	}
+
+	uniqueMethods := make([]string, 0)
+	for m := range httpMethods {
+		uniqueMethods = append(uniqueMethods, m)
+	}
+
+	return uniqueMethods
+}
+
+func (p *Plugin) GetRequiredMethods(options []av_opts.AtlasValidateFieldOption_Operation) []string {
+	requiredMethods := make(map[string]struct{}, 0)
+	for _, op := range options {
+		switch op {
+		case av_opts.AtlasValidateFieldOption_create:
+			requiredMethods["POST"] = struct{}{}
+		case av_opts.AtlasValidateFieldOption_update:
+			requiredMethods["PATCH"] = struct{}{}
+			requiredMethods["PUT"] = struct{}{}
+		}
+	}
+
+	uniqueMethods := make([]string, 0)
+	for m := range requiredMethods {
+		uniqueMethods = append(uniqueMethods, m)
+	}
+
+	return uniqueMethods
+}
+
+func (p *Plugin) generateValidateRequired(md *descriptor.DescriptorProto, t string) {
+	requiredFields := make(map[string][]string)
+	for _, fd := range md.GetField() {
+		if fExt, err := proto.GetExtension(fd.Options, av_opts.E_Field); err == nil && fExt != nil {
+			favOpt := fExt.(*av_opts.AtlasValidateFieldOption)
+			methods := p.GetRequiredMethods(favOpt.GetRequired())
+			if len(methods) == 0 {
+				continue
+			}
+			requiredFields[fd.GetName()] = methods
+		}
+	}
+
+	p.P(fmt.Sprintf(`func ValidateRequired_Object_%s(ctx context.Context, v map[string]json.RawMessage) error {`, t))
```

Comment:
Don't think that this method should be exported, It's just a matter of taste, but I would avoid mixing camelCase for what we can control (i mean object name can be in any format and case, but it's expected that we follow one convention in method naming). call it 'validate_required_Object_%s'